pkg/resources/ingresses/ingress.go

```diff
@@ -82,7 +80,10 @@ func NewIngressForCRWithSSL(existing *netv1.Ingress, namespacedName types.Namesp
 
 	desired.Spec.Rules[0].Host = host
 	if sslEnabled {
-		desired.Annotations = map[string]string{"nginx.ingress.kubernetes.io/ssl-passthrough": "true"}
+		if desired.Annotations == nil {
+			desired.Annotations = make(map[string]string)
+		}
+		desired.Annotations["nginx.ingress.kubernetes.io/ssl-passthrough"] = "true"
```

Comment:
The annotation `nginx.ingress.kubernetes.io/ssl-passthrough` is specific for nginx ingress controller and the new templates field could be used to set this annotation. What about removing the annotation `nginx.ingress.kubernetes.io/ssl-passthrough` and adding to the doc how to set it using the new templates field?

Reply:
I was thinking that could be a follow up commit, it would be a good example of usage. I like the idea. Thanks.